src/index.js

```diff
@@ -1,35 +1,73 @@
-const core = require('@actions/core');
-const github = require('@actions/github');
+const core = require("@actions/core");
+const github = require("@actions/github");
 
 async function main() {
-  const { eventName, sha, ref, repo: { owner, repo }, payload } = github.context;
+  const {
+    sha,
+    ref,
+    repo: { owner, repo },
+    payload
+  } = github.context;
+
+  // GITHUB_RUN_ID is provided by Github
+  const run_id = process.env.GITHUB_RUN_ID;
+
   let branch = ref.slice(11);
   let headSha = sha;
+
   if (payload.pull_request) {
     branch = payload.pull_request.head.ref;
     headSha = payload.pull_request.head.sha;
   }
 
-  console.log({ eventName, sha, headSha, branch, owner, repo });
-  const workflow_id = core.getInput('workflow_id', { required: true });
-  const token = core.getInput('access_token', { required: true });
-  console.log(`Found input: ${workflow_id}`);
-  console.log(`Found token: ${token ? 'yes' : 'no'}`);
+  const token = core.getInput("access_token", { required: true });
+
   const octokit = new github.GitHub(token);
-  const { data } = await octokit.actions.listWorkflowRuns({ owner, repo, workflow_id, branch });
-  console.log(`Found ${data.total_count} runs total.`);
-  const runningWorkflows = data.workflow_runs.filter(
-    workflow => workflow.head_sha !== headSha && workflow.status !== 'completed'
+
+  // Obtain workflow id from url
+  // Example url: https://api.github.com/repos/octo-org/octo-repo/actions/workflows/30433642
+  const {
+    data: { workflow_url }
+  } = await octokit.actions.getWorkflowRun({ owner, repo, run_id });
+  
+  const workflow_id = workflow_url.substring(
+    workflow_url.lastIndexOf("/workflows/") + 11,
+    Number.POSITIVE_INFINITY
+  );
+
+  // Obtain workflow runs
+  const { data: runs } = await octokit.actions.listWorkflowRuns({
+    owner,
+    repo,
+    workflow_id,
+    branch
+  });
+
+  core.info(`Found ${runs.total_count} runs total.`);
+
+  // Filter workflow runs
+  const runningWorkflows = runs.workflow_runs.filter(
+    workflow => workflow.head_sha !== headSha && workflow.status !== "completed"
```

Comment:
`workflow` is actually not a _workflow_ but a _workflow run_, so we should name it accordingly. Also please use single quotes instead of double quotes:
```suggestion
    workflowRun => workflowRun.head_sha !== headSha && workflowRun.status !== 'completed'
```